exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -72,7 +74,17 @@ func (q *boundedMemoryQueue) Produce(item Request) bool {
 	// their combined size is stored in q.size, and their combined capacity
 	// should match the capacity of the new queue
 	if q.size.Load() >= q.capacity {
-		return false
+		if q.overflow == nil {
+			return false
+		}
+		select {
+		case old := <-q.items:
+			q.size.Add(^uint32(0))
+			q.overflow(old)
```

Comment:
Can you please clarify why this is needed?

Reply:
If the queue is full and a new request is added to the queue, the head of the queue will be removed and sent through the overflow callback. This prioritizes live requests in the primary queue and relegates older requests and retry requests to the backlog in the case of queue overflow.